### DISCUSSION TITLE
duplicate toolbar in dialog

### DISCUSSION THREAD

**Discussion by juliermefelix** [serenity-is/Serenity#6443] [Help] [answered]
Dear,
I would like to duplicate toolbar so that it is also displayed at the bottom of the dialog.
Keep the default toolbar and another copy at the bottom of the dialog.

![image](https://user-images.githubusercontent.com/7516443/176952479-4697a1d1-0721-4d15-a082-7a83bfbe94a5.png)

I was only able to move to the bottom.
![image](https://user-images.githubusercontent.com/7516443/176952570-244b7a3f-fe35-489b-bb90-a1ae90cc4916.png)

**Comment by dfaruque** [comment, marked as answer]
place the following code in yourDialog.ts

        toolbarBottom: Serenity.Toolbar;

        protected initToolbar() {
            super.initToolbar();

            let TooltarBottomDiv = this.byId('ToolbarBottom');
            if (TooltarBottomDiv.length === 0) {
                return;
            }
            let opt = { buttons: this.getToolbarButtons() };
            this.toolbarBottom = new Serenity.Toolbar(TooltarBottomDiv, opt);
        }

        findButton(className) {
            let buttons = this.toolbar.findButton(className)[0];
            let bottonButtons = this.toolbarBottom.findButton(className)[0];

            return jQuery([buttons, bottonButtons]);
        }

        syncToolButtons() {
            this.toolbarBottom.updateInterface();

            var bottonToolButtons = this.toolbarBottom.element.find('.tool-button');
            this.toolbar.element.find('.tool-button')
                .each((i, e) => {
                    for (let j = 0; j < e.attributes.length; j++) {
                        let attr = e.attributes[j];
                        bottonToolButtons[i].setAttribute(attr.name, attr.value);
                    }
                });
        }

        protected afterLoadEntity() {
            super.afterLoadEntity();

            this.syncToolButtons();
        }

        protected getTemplate() {
            return `<div class="s-DialogContent">
                        <div id="~_Toolbar" class="s-DialogToolbar"></div>
                        <div class="s-Form">
                            <form id="~_Form" action="">
                                <div class="fieldset ui-widget ui-widget-content ui-corner-all">
                                    <div id="~_PropertyGrid"></div>
                                    <div class="clear"></div>
                                </div>
                            </form>
                        </div>
                        <br />
                        <div id="~_ToolbarBottom" class="s-DialogToolbar"></div>
                    </div>`;
        }

**Reply by juliermefelix** [reply]
Thanks dfaruque

![image](https://user-images.githubusercontent.com/7516443/177045157-cfcce15c-dc52-4c6c-99e1-1db9d4d59d22.png)

**Reply by stixoffire** [reply]
So totally awesome - and the timing is perfecto ...
**Thanks very much dfaruque !**
